lxd/daemon.go

```diff
@@ -755,15 +764,48 @@ func (d *Daemon) init() error {
 		candidExpiry = config.CandidExpiry()
 		candidDomains = config.CandidDomains()
 		maasAPIURL, maasAPIKey = config.MAASController()
+		rbacAPIURL = config.RBACAPIURL()
+		rbacAPIKey = config.RBACAPIKey()
+		rbacAgentPrivateKey = config.RBACAgentPrivateKey()
+		rbacAgentPublicKey = config.RBACAgentPublicKey()
+		rbacAPIUsername = config.RBACAPIUsername()
+		rbacExpiry = config.RBACAPIExpiry()
+		rbacAuthURL = config.RBACAuthURL()
+
 		return nil
 	})
 	if err != nil {
 		return err
 	}
 
-	err = d.setupExternalAuthentication(candidEndpoint, candidEndpointKey, candidExpiry, candidDomains)
-	if err != nil {
-		return err
+	// Ensure that not both Candid and RBAC are configured at the same time
+	if candidEndpoint != "" && rbacAPIURL != "" {
+		logger.Warn("Disabling Candid and RBAC since they cannot be configured simultaneously")
+	} else {
+		if rbacAPIURL != "" {
+			err = d.setupExternalAuthentication(rbacAuthURL, "", rbacExpiry, "")
+			if err != nil {
+				return err
+			}
+
+			go func() {
+				for {
+					err = d.setupRBACServer(rbacAPIURL, rbacAPIKey, rbacAPIUsername, rbacAgentPrivateKey, rbacAgentPublicKey, rbacAuthURL)
+					if err == nil {
+						logger.Info("Successfully set up RBAC", log.Ctx{"url": rbacAPIURL})
+						break
+					}
+
+					logger.Warn("Unable to set up RBAC, trying again in a minute", log.Ctx{"url": rbacAPIURL, "err": err})
+					time.Sleep(time.Minute)
```

Comment:
Current logic is a very bad idea as this means that we'd have candid setup but no RBAC, effectively allowing users to do whatever they want until rbac is up.